kubernetes-addons/Observability/README.md

```diff
@@ -0,0 +1,137 @@
+# How-To Setup Observability for OPEA Workload in Kubernetes
+
+This guide provides a step-by-step approach to setting up observability for the OPEA workload in a Kubernetes environment. We will cover the setup of Prometheus and Grafana, as well as the collection of metrics for Gaudi hardware, OPEA/chatqna including TGI,TEI-Embedding,TEI-Reranking and other microservies, and PCM.
+
+## 1. Setup Prometheus & Grafana
+
+Setting up Prometheus and Grafana is essential for monitoring and visualizing your workloads. Follow these steps to get started:
+
+### Step 1: Install Prometheus&Grafana
+
+```
+kubectl create ns monitoring
+helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+helm repo update
+helm install prometheus-stack prometheus-community/kube-prometheus-stack --version 55.5.1 -n monitoring
+```
+
+### Step 2: Verify the installation:
+
+```
+kubectl get pods -n monitoring
+```
+
+### Step 3: Port-forward to access Grafana:
+
+```
+kubectl port-forward service/grafana 3000:80
+```
+
+### Step 4: Access Grafana:
+
+Open your browser and navigate to http://localhost:3000. Use "admin/prom-operator" as the username and the password to login.
+
+## 2. Metric for Gaudi Hardware(v1.16.2)
+
+To monitor Gaudi hardware metrics, you can use the following steps:
+
+### Step 1: Install daemonset
+
+```
+kubectl create -f https://vault.habana.ai/artifactory/gaudi-metric-exporter/yaml/1.16.2/metric-exporter-daemonset.yaml
+```
+
+### Step 2: Install metric-exporter
+
+```
+kubectl create -f https://vault.habana.ai/artifactory/gaudi-metric-exporter/yaml/1.16.2/metric-exporter-service.yaml
+```
+
+### Step 3: Install service-monitor
+
+```
+kubectl apply -f ./habana/metric-exporter-serviceMonitor.yaml
+```
+
+### Step 4: Verify the metrics
+
+```
+# To get the metric endpoints
+kubectl get ep -n monitoring |grep metric-exporter
+# Fetch the metrics
+curl ${metric_exporter_ip}:41611/metrics
```

Comment:
This could use something similar to: https://github.com/opea-project/GenAIInfra/tree/main/helm-charts#verify-hpa-metrics

Reply:
the habana metric endpoints is a headless service, but still I already refered your suggestion to modify it.